src/app/static/src/js/components/runWorkflow/runWorkflowCreate.vue

```diff
@@ -1,44 +1,172 @@
+                    class="btn btn-success"
+                    :disabled="enableButtons">Re-run workflow
+            </button>
+            <button id="clone" @click="clone()"
+                    type="button" class="btn btn-success"
+                    :disabled="enableButtons">Clone workflow
+            </button>
+        </div>
+        <div class="pt-4 col-sm-6">
+            <error-info :default-message="defaultMessage" :api-error="error"></error-info>
         </div>
     </div>
 </template>
 
 <script lang="ts">
-import Vue from "vue"
+    import Vue from "vue"
+    import vSelect from 'vue-select'
+    import {api} from "../../utils/api";
+    import {RunWorkflowMetadata} from "../../utils/types";
+    import {longTimestamp} from "../../utils/helpers";
+    import ErrorInfo from "../errorInfo.vue";
+
+    interface Methods {
+        create: () => void
+        clone: () => void
+        rerun: () => void
+        getWorkflows: () => void
+        getCloneableWorkflow: () => void
+        searchWorkflows: (options: [], search: string) => void
+        getLongTimestamp: (date: string) => string
+        getUsername: (email: string) => string
+    }
+
+    interface Data {
+        workflows: []
+        error: string | null
+        defaultMessage: string | null
+        selectedWorkflow: string
+        runWorkflowMetadata: RunWorkflowMetadata | null
+    }
 
-interface Methods {
-    create: () => void
-    clone: () => void
-    rerun: () => void
-}
+    let cloneableWorkflowMetadata = {
+        name: null,
+        date: null,
+        email: null,
+        reports: {},
+        instances: {},
+        git_branch: null,
+        git_commit: null,
+        key: null
+    }
+
+    interface Computed {
+        enableButtons: boolean
+    }
 
-export default Vue.extend<unknown, Methods, unknown, unknown>({
-    name: "runWorkflowCreate",
-    methods: {
-        create: function () {
-            this.$emit("create")
+    export default Vue.extend<Data, Methods, Computed, unknown>({
+        name: "runWorkflowCreate",
+        data(): Data {
+            return {
+                workflows: [],
+                error: "",
+                defaultMessage: "",
+                selectedWorkflow: null,
+                runWorkflowMetadata: null
+            }
+        },
+        computed: {
+            enableButtons: function () {
+                return !this.selectedWorkflow || !this.runWorkflowMetadata
+            }
         },
-        clone: function () {
-            this.$emit("clone")
+        methods: {
+            create: function () {
+                this.$emit("create")
+            },
+            clone: function () {
+                if (this.selectedWorkflow && this.runWorkflowMetadata) {
+                    const {reports, git_branch, git_commit} = this.runWorkflowMetadata
+                    const clonedWorkflow = {...cloneableWorkflowMetadata, ...{reports, git_branch, git_commit}}
```

Comment:
I actually think that it would be better if this page (and each page of wizard) made as few assumptions as possible about the structure of a workflow. So this function simply returns `{reports, git_branch, git_commit}` to `runWorkflow`. Otherwise if we subsequently add extra fields we have to remember to find every variable similar to `cloneableWorkflowMetadata` and ensure that it's updated.